### ISSUE TITLE
Use 'override' keyword for reimplemented virtual functions

### ISSUE THREAD

**Issue by StefanBruens** [taglib/taglib#865]
Since C++11, reimplemented functions should be marked with 'override'.

This turns silent runtime errors into compile time errors:

```
class A {
public:
  virtual int foo();
  virtual int bar();
};

class B : public A {
public:
  virtual int foo();
  virtual int baz();   <- typo, should be bar()
};
```
`B b; b->bar();` silently calls A::bar(), although there is (should be) a reimplementation in B.

For backwards compatibility:
```
#if __cplusplus < 201103L
#define TAGLIB_OVERRIDE
#else
#define TAGLIB_OVERRIDE override
#endif
```
`virtual int baz() TAGLIB_OVERRIDE;` now creates a compile time error - there is no `virtual int baz() `in A.

**Comment by StefanBruens** [comment]
Why is this tagged with 'taglib2'? Adding `override` is a completely API and ABI compatible change.

**Comment by scotchi** [comment]
Because master is still C++98.

**Comment by StefanBruens** [comment]
... and with C++98, `TAGLIB_OVERRIDE` expands to an empty string. But when you use C++11 (which all contemporary compilers default to), you get some additional useful diagnostics.

**Comment by sbooth** [comment]
Fixed by 9867bc9
